### PR TITLE
[contrib:redis] marking redis service as CACHE by default

### PR DESCRIPTION
**Pull request by ufoot**

### CODE REVIEW

File: lib/ddtrace/contrib/redis/patcher.rb [hunk 1]
```diff
@@ -59,7 +59,7 @@ def patch_redis_client
             end
 
             def initialize(*args)
-              pin = Datadog::Pin.new(SERVICE, app: 'redis', app_type: Datadog::Ext::AppTypes::CACHE)
+              pin = Datadog::Pin.new(SERVICE, app: 'redis', app_type: Datadog::Ext::AppTypes::DB)
```
Comment (palazzem) [inline lib/ddtrace/contrib/redis/patcher.rb]: did we choose when we should use the `cache` type?

Comment (ufoot) [inline lib/ddtrace/contrib/redis/patcher.rb]: This should ultimately be the user to decide about. Redis can act as both, this is just a default value. Most of the "added value" of the patch here is: actually do use the value in the Pin and avoid redis spans being reported as "custom" (which is clearly not the case).

File: lib/ddtrace/contrib/redis/patcher.rb [hunk 2]
```diff
@@ -60,6 +61,9 @@ def patch_redis_client
             def initialize(*args)
               pin = Datadog::Pin.new(SERVICE, app: 'redis', app_type: Datadog::Ext::AppTypes::DB)
               pin.onto(self)
+              if pin.tracer && pin.service && pin.app && pin.app_type
```
Comment (LotharSee) [inline lib/ddtrace/contrib/redis/patcher.rb]: Do you really need to test for `pin.app && pin.app_type`?

File: lib/ddtrace/contrib/redis/patcher.rb (hunk 1 above)
Comment (LotharSee) [inline lib/ddtrace/contrib/redis/patcher.rb]: Our "service type" still lack a real Spec, but for now let's just have it consistent with Python. So here, gtm!

Comment (palazzem) [inline lib/ddtrace/contrib/redis/patcher.rb]: 👍

File: lib/ddtrace/contrib/redis/patcher.rb (hunk 2 above)
Comment (ufoot) [inline lib/ddtrace/contrib/redis/patcher.rb]: Since the pin is created just before and we explicitly set those, no. Good catch.

**Pull request merged**